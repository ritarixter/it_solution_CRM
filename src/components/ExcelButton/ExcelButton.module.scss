@import "../../styles/variables.scss";
.excel {
  color: $mainColor;
  font-size: 13px;
  margin: 0;
  cursor: pointer;
  transition: opacity 0.2s ease;
  display: flex;
  align-items: center;
  &__icon {
    margin-right: 3px;
  }
  &:hover {
    opacity: 0.8;
  }
}
