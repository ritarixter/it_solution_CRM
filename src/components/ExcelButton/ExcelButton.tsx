import { FC } from "react";
import styles from "./ExcelButton.module.scss";
import excel from "../../images/icons/excel_icon.svg";

export const ExcelButton: FC<{ onClick: () => void }> = ({onClick}) => {
  return (
    <p className={styles.excel} onClick={onClick}>
      <img src={excel} className={styles.excel__icon} alt="Иконка excel" />
      Экспорт
    </p>
  );
};
