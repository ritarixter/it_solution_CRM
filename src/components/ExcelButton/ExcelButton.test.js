import React from 'react';
import renderer from 'react-test-renderer';

import { ExcelButton } from './ExcelButton';

it('ExcelButton рендерится без ошибок', () => {
  const tree = renderer
    .create(<ExcelButton text="Импорт" onClick={()=>{}} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
}); 