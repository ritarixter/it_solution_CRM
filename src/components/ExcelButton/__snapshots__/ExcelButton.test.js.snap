// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExcelButton рендерится без ошибок 1`] = `
<p
  className="excel"
  onClick={[Function]}
>
  <img
    alt="Иконка excel"
    className="excel__icon"
    src="excel_icon.svg"
  />
  Импорт
</p>
`;
